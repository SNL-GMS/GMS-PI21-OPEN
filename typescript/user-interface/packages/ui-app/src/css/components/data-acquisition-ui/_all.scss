@import 'soh-vars';
@import 'soh-cell';
@import 'ag-theme-overrides.scss';
@import 'data-acquisition';
@import 'environment-history';
@import 'historical-trends-history';
@import 'soh-chart';
@import 'station-statistics';
@import 'soh-lag-missing';
@import 'soh-environment';
@import 'soh-overview';
@import 'soh-quiet-indicator';
@import 'capability-counts';
@import 'soh-map';
