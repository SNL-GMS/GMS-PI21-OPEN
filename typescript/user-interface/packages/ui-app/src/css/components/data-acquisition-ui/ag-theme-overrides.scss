.ag-theme-dark--soh {
  border-radius: $cell-cap-radius;

  .ag-header {
    background: transparent;
  }

  .ag-header-cell {
    .ag-react-container {
      width: 100%;
    }
  }
  .ag-header-viewport {
    background-color: var(--gms-prominent-background, $gms-prominent-background);
  }

  .ag-row {
    overflow: hidden;
    border-radius: $cell-cap-radius;
  }

  .ag-ltr .ag-cell:not(.ag-cell-focus).ag-cell-last-left-pinned,
  .ag-ltr .ag-row.ag-cell-last-left-pinned,
  .ag-horizontal-left-spacer {
    border-right: none;
  }

  &:not(.table--cell-selection-only) {
    .ag-row-hover .ag-cell {
      @include brighten;
    }
  }
  &.table--cell-selection-only {
    .ag-cell:hover {
      @include brighten;
    }
  }

  // TODO: Make this dynamic so it doesn't expect a fixed
  // number of columns
  .ag-row-selected {
    .ag-cell .table-cell {
      box-sizing: border-box;
      @include selectedBorder('top', 'bottom');
    }
    .ag-cell[aria-colindex='1'] .table-cell {
      @include selectedBorder('left');
    }
    .ag-cell .table-cell[data-col-position='last'] {
      @include selectedBorder('right');
    }
  }

  .ag-body-viewport,
  .ag-root-wrapper {
    background: transparent;
  }
}
