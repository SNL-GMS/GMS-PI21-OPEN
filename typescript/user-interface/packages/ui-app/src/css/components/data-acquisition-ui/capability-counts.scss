.stacked-counts {
  @include flexColumn;
  @include centerChildren;
  &__entry {
    padding: 0;
    margin: 0;
    line-height: 1em;
    padding-right: 0.25rem;
    margin-left: -0.25rem;
    font-weight: 400;
    opacity: 0.8;
    font-size: 0.875em;
    & + & {
      margin-top: 0.125rem;
    }
  }
}

// Color the marginal capability count yellow if it is inside of a red cell
[data-cell-status='bad'] .stacked-counts__entry[data-capability-status='marginal'] {
  color: $marginal-color;
}
