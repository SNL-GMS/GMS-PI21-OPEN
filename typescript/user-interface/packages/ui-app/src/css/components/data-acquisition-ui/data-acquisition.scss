.form-field-small {
  width: 60%;
}

.form-field-large {
  margin-left: 5%;
  width: 80%;
}

.footer-button-row {
  position: absolute;
  top: 90%;
  left: 1.5%;
}

.footer-button {
  position: relative;
  display: inline;
}

.footer-button-middle {
  position: relative;
  display: inline;
  left: 200%;
}

.footer-button-right {
  position: relative;
  display: inline;
  left: 225%;
}

.station-configuration-form__row {
  display: flex;
  flex-direction: row;
  margin-bottom: 1%;
}

.station-configuration-form {
  display: flex;
  position: absolute;
  top: 0%;
  height: 20%;
  right: 0%;
  left: 0.5%;
  flex-direction: column;
  width: 100%;
  overflow-y: scroll;
}

.station-configuration-inner-menu {
  position: absolute;
  top: 45%;
  height: 20%;
  right: 0%;
  left: 1.5%;
}

.station-configuration-bottom-menu {
  position: absolute;
  top: 68%;
  height: 20%;
  right: 0%;
  left: 1.5%;
}

.station-configuration-form__labels-top-left {
  margin-left: 1% !important;
  margin-top: 0.5% !important;
  margin-right: 1% !important;
  font-size: small;
}

.station-configuration-form__labels-top-right {
  margin-left: 1% !important;
  margin-top: 1% !important;
  font-size: small;
}

.station-configuration-form__labels-middle {
  margin-left: 1% !important;
  margin-right: 15px !important;
  font-size: small;
}

.station-configuration-form__field-top {
  margin-left: 1.5% !important;
  margin-top: 15px !important;
  font-size: small;
}

.station-configuration-form__field-middle {
  margin-left: 12px !important;
  margin-top: 5px !important;
  font-size: small;
  width: 200%;
}

.station-configuration-form__field-top {
  margin-left: 5px !important;
  margin-top: 5px !important;
  font-size: small;
}

.station-configuration-form__field-bottom {
  margin-left: 3px !important;
  margin-top: 5px !important;
  font-size: small;
}

.station-configuration-form__labels-bottom-right {
  margin-left: 1% !important;
  font-size: small;
}

.station-configuration-form__labels-bottom-left {
  margin-left: 1% !important;
  font-size: small;
}

.station-configuration-form__input-middle {
  padding-left: 10% !important;
  width: 200%;
}

.configure-station-groups-select-network {
  position: relative;
  top: 0%;
  height: 100%;
  width: 40%;
  right: 0%;
  left: 0%;
}

.configure-station-groups-select-network-table {
  position: relative;
  top: 0%;
  height: 100%;
  max-height: calc(100% - 49px);
  width: 100%;
  right: 0%;
  left: 0%;
}

.configure-station-groups-edit-stations {
  position: absolute;
  top: 0%;
  height: 98%;
  right: 0%;
  left: 41%;
  visibility: hidden;
}

.configure-station-groups-no-network-selected {
  position: absolute;
  top: 0%;
  height: 98%;
  right: 0%;
  left: 41%;
  visibility: visible;
}

.configure-station-groups-associated-stations {
  position: absolute;
  top: 0%;
  height: 100%;
  width: 45%;
  right: 0%;
  left: 0%;
}

.configure-station-groups-arrows {
  position: absolute;
  top: 40%;
  height: 100%;
  width: 4%;
  right: 0%;
  left: 48%;
}

.configure-station-groups-available-stations {
  position: absolute;
  top: 0%;
  height: 100%;
  width: 45%;
  right: 0%;
  left: 55%;
}

.data-acquisition-popover {
  font-size: small !important;
  max-height: 300px !important;
  overflow-y: auto !important;
  margin: 0.5rem;
  display: inline-table;
}

.data-acquisition-import-css-popover-with-table {
  font-size: small !important;
  max-height: 500px !important;
  overflow-y: scroll !important;
  margin: 0.5rem;
}

.data-acquisition-inputs input {
  border: 1px solid var(--gms-main, $gms-main);
}

.data-acquisition-inputs input:focus {
  outline: none !important;
  border: 1px solid $data-acq-strong-warning;
}

.data-acquisition-cell .ag-cell {
  display: flex;
  justify-content: center;
  align-items: center !important;
}

.soh-update-time-controls {
  flex: 0 0 auto;
  width: 100%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  color: var(--gms-main, $gms-main) !important;
}

.soh-update-time-controls__label {
  margin-bottom: 0rem !important;
  margin-left: 1rem !important;
  color: var(--gms-main, $gms-main);
}

.soh-update-time {
  font-family: $gms-mono;
  color: var(--gms-main, $gms-main);
}

.text-opacity-80 {
  opacity: 0.8;
}

.soh-drill-down-station-label {
  font-size: 1.5em;
  margin-top: 0.25em;
  padding-left: 0.125em;
  text-rendering: geometricPrecision;
}

.display-title {
  padding: 0.25em 0 0 0.25em;
  display: flex;
  align-items: baseline;
  &__subtitle {
    margin-left: 0.5em;
    color: var(--gms-prominent, $gms-prominent);
    font-size: 1.1rem;
    font-weight: 100;
    @include ellipsis;
  }
}

.soh-table-label {
  padding: 0.5em 0 0.25em 0;
  color: var(--gms-prominent, $gms-prominent);
  margin-left: 0.25em;
  font-size: 1.1rem;
  font-weight: 400;
  font-family: $gms-sans;
  -webkit-font-smoothing: antialiased;
}

.drop-zone__wrapper .drop-zone {
  &--highlighted {
    background-color: var(--gms-table-selection, $gms-table-selection);
  }
}
