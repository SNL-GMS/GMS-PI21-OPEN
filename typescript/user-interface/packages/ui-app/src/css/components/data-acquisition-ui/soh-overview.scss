$badge-padding: 6px;

.soh-overview {
  height: 100%;
  user-select: none;
  overflow: auto;
  .divider-container {
    .bottom-component {
      display: flex;
    }
  }
  .deselect-handler {
    height: calc(100% - #{$toolbar-height} - 0.5rem);
  }
}

/// Toolbar
.soh-overview-toolbar__container {
  margin-bottom: 0.5rem;
}

/// Group header
.soh-overview-group-header {
  background-color: var(--gms-recessed, $gms-recessed);
  &__container {
    display: grid;
    grid-template-columns: 5fr 1fr 1fr 1fr;
    font-size: 24px;
  }
  &__network {
    padding: 6px;
    margin-right: 12px;
    border-radius: $rounded-rectangle-radius;
    color: var(--gms-main, $gms-main);
    background-color: var(--gms-prominent-background, $gms-prominent-background);
    &[data-capability-status='bad'] {
      background: $bad-color;
    }
    &[data-capability-status='marginal'] {
      background: $marginal-color;
      color: var(--gms-background, $gms-background);
    }
    &[data-capability-status='good'] {
      background: $good-color;
      color: var(--gms-main, $gms-main);
    }
    img.badge {
      @include addBoxShadow('md');
      float: right;
    }
  }
  &__count {
    display: flex;
    align-items: center;
    justify-content: center;
    font-family: $gms-sans;
    font-weight: bold;
    text-align: center;
  }
  &__count--bad {
    color: lighten($gms-strong-warning, 10%);
  }
  &__count--marginal {
    color: darken($gms-warning, 10%);
  }
  &__count--good {
    color: lighten($gms-good, 10%);
  }
  &__count--none {
    color: lighten($gms-background, 20%);
  }
}

/// Table
.soh-overview-table {
  display: flex;
  flex-direction: column;
  height: 100%;
  width: 100%;
  &__wrapper {
    background-color: var(--gms-popover-background, $gms-popover-background);
    padding: 4px;
    border-radius: 4px;
    display: flex;
    flex-direction: column;
    & .top-component {
      transition: height ease 0.08s;
    }
  }
  &__bin {
    min-height: 50px;
    display: flex;
    flex-direction: column;
    flex-grow: 1;
    &--top {
      min-height: 50px;
      flex-grow: 1;
    }
    &--bottom {
      height: 100%;
    }
  }
  &__label {
    font-size: 1.25em;
    letter-spacing: 0.5px;
    border-radius: 4px;
    display: flex;
    padding-left: 0.25em;
    margin-top: 0.25em;
    color: var(--gms-prominent, $gms-prominent);
    align-content: center;
  }

  /// Drop zone highlighting
  & &__drop-zone,
  & .soh-overview-table__drop-zone {
    height: 100%;
    flex-grow: 1;
    &--highlighted {
      background-color: var(--gms-table-selection, $gms-table-selection);
      pointer-events: all !important;
    }
  }
}

/// Cells

.soh-overview-cell {
  $cell-height: 30px;
  $cell-margin: 12px;
  @include gpuAccelerated;
  letter-spacing: 1.25px;
  font-size: 14px;
  font-weight: 500;
  padding: 0 0.375em 0 0.75em;
  height: $cell-height;
  display: flex;
  justify-content: left;
  align-items: center;
  margin-right: $cell-margin;
  border-radius: 4px;
  margin-bottom: $cell-margin;
  @include disabledBackground;
  cursor: pointer;
  &:hover {
    @include brighten;
  }
  &__container {
    position: relative;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(13ch, 1fr));
    grid-template-rows: repeat(auto-fill, #{$cell-height + $cell-margin});
    padding-top: $badge-padding + 4;
    padding-left: calc($badge-padding / 2);
    padding-right: 2px;
    flex-grow: 1;
    overflow-y: auto;
    overflow-y: overlay;
    background-color: var(--gms-prominent-background, $gms-prominent-background);
    border-radius: 4px;
    @include withScrollSnap(y, mandatory, start, 0.5em);
  }
  &[data-soh-status='good'] {
    background: $good-color;
  }
  &[data-soh-status='marginal'] {
    background: $marginal-color;
    color: var(--gms-background, $gms-background);
  }
  &[data-soh-status='bad'] {
    background: $bad-color;
  }

  @include createHashedSohCellStyles('[data-soh-status="none"]', $non-contributing-color);
  @include createHashedSohCellStyles('[data-soh-status=""]', $non-contributing-color);

  &--selected {
    @include selectedBorder;
    &::after {
      border-radius: $rounded-rectangle-radius;
    }
    // Override the selected border to account for the added width of the 1 px border that these cells have
    &[data-soh-status='none'],
    &[data-soh-status=''] {
      border: none;
    }
  }
  // if a table is draggable, give it a hover state and cursor to indicate as much
  &--draggable {
    cursor: pointer;
  }
  // Compensate for the soh-overview-cell's padding in this case
  .soh-cell__right-container {
    margin-right: -0.375em;
  }
}

.resize-container {
  .resizer {
    transition: opacity 120ms ease;
  }
  &--resizing .resizer {
    opacity: 0.5;
  }
  &--resizing .resizer--resizing {
    opacity: 1;
  }
}
