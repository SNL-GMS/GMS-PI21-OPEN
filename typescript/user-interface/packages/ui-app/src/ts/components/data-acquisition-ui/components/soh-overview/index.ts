export { SohOverviewContainer as SohOverview } from './soh-overview-container';
