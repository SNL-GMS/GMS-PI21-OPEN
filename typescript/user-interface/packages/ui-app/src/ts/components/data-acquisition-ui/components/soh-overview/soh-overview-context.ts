import type { SohTypes } from '@gms/common-model';
import type GoldenLayout from '@gms/golden-layout';
import React from 'react';

/**
 * The type for the data used by the context
 */
export interface SohOverviewContextData {
  glContainer: GoldenLayout.Container;
  stationGroupSoh: SohTypes.StationGroupSohStatus[];
  stationSoh: SohTypes.UiStationSoh[];
  selectedStationIds: string[];
  sohStationStaleTimeMS: number;
  updateIntervalSecs: number;
  quietTimerMs: number;
  setSelectedStationIds(ids: string[]): void;
  acknowledgeSohStatus(stationIds: string[], comment?: string): void;
}

/**
 * Instantiate the Context and set up the defaults
 */
export const SohOverviewContext: React.Context<SohOverviewContextData> = React.createContext<
  SohOverviewContextData
>(undefined);
