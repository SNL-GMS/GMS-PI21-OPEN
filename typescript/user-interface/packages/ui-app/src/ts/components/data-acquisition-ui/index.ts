import * as DataAcquisitionComponents from './components';

export { DataAcquisitionComponents };
