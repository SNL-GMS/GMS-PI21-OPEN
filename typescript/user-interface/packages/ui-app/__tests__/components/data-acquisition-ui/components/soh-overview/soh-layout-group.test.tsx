import { SohTypes } from '@gms/common-model';
import { uuid } from '@gms/common-util';
import { render } from '@testing-library/react';
import React from 'react';

import type { SohOverviewContextData } from '../../../../../src/ts/components/data-acquisition-ui/components/soh-overview/soh-overview-context';
import { SohOverviewContext } from '../../../../../src/ts/components/data-acquisition-ui/components/soh-overview/soh-overview-context';
import {
  generateSohStationGroupMap,
  StationGroupsLayout
} from '../../../../../src/ts/components/data-acquisition-ui/components/soh-overview/station-groups/station-groups-layout';
import { stationAndStationGroupSohStatus } from '../../../../__data__/data-acquisition-ui/soh-overview-data';

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

// eslint-disable-next-line import/no-deprecated
const lodash = jest.requireActual('lodash');
lodash.uniqueId = () => '1';

const contextValues: SohOverviewContextData = {
  stationSoh: stationAndStationGroupSohStatus.stationSoh,
  stationGroupSoh: stationAndStationGroupSohStatus.stationGroups,
  acknowledgeSohStatus: jest.fn(),
  glContainer: undefined,
  quietTimerMs: 5000,
  updateIntervalSecs: 5,
  selectedStationIds: [],
  setSelectedStationIds: jest.fn(),
  sohStationStaleTimeMS: 30000
};

const statusesToDisplay: SohTypes.SohStatusSummary[] = [
  SohTypes.SohStatusSummary.GOOD,
  SohTypes.SohStatusSummary.MARGINAL,
  SohTypes.SohStatusSummary.BAD
];

describe('Soh station groups layout', () => {
  let idCount = 0;
  // eslint-disable-next-line no-plusplus
  uuid.asString = jest.fn().mockImplementation(() => ++idCount);
  it('should be defined', () => {
    expect(StationGroupsLayout).toBeDefined();
  });

  it('matches the snapshot when provided with a context', () => {
    const { container } = render(
      <SohOverviewContext.Provider value={contextValues}>
        <StationGroupsLayout
          statusesToDisplay={statusesToDisplay}
          isHighlighted
          stationGroupsToDisplay={contextValues.stationGroupSoh.map(
            stationGroup => stationGroup.stationGroupName
          )}
        />
      </SohOverviewContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('can generate the SohStationGroupMap', () => {
    const stationGroupMap = generateSohStationGroupMap(
      true,
      contextValues.stationGroupSoh,
      contextValues.stationSoh,
      statusesToDisplay
    );
    expect(stationGroupMap).toBeDefined();
    expect(stationGroupMap).toMatchSnapshot();
  });

  it('matches the snapshot when provided with no station groups to display', () => {
    const { container } = render(
      <SohOverviewContext.Provider value={contextValues}>
        <StationGroupsLayout
          statusesToDisplay={statusesToDisplay}
          isHighlighted
          stationGroupsToDisplay={[]}
        />
      </SohOverviewContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
