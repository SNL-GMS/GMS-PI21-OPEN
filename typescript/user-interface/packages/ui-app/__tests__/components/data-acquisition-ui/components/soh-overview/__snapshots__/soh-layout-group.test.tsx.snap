// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Soh station groups layout can generate the SohStationGroupMap 1`] = `
Map {
  "Group 4" => Array [
    Object {
      "allStationAggregates": Array [],
      "channelSohs": Array [],
      "id": "1",
      "needsAcknowledgement": true,
      "needsAttention": true,
      "sohStatusSummary": "GOOD",
      "stationGroups": Array [
        Object {
          "groupName": "Group 4",
          "sohStationCapability": "GOOD",
          "stationName": "H05N",
        },
      ],
      "stationName": "H05N",
      "statusContributors": Array [
        Object {
          "contributing": true,
          "statusSummary": "GOOD",
          "type": "LAG",
          "value": 5,
          "valuePresent": true,
        },
      ],
      "time": 2,
      "uuid": "1",
    },
    Object {
      "allStationAggregates": Array [],
      "channelSohs": Array [],
      "id": "2",
      "needsAcknowledgement": true,
      "needsAttention": true,
      "sohStatusSummary": "MARGINAL",
      "stationGroups": Array [
        Object {
          "groupName": "Group 4",
          "sohStationCapability": "GOOD",
          "stationName": "H06N",
        },
      ],
      "stationName": "H06N",
      "statusContributors": Array [
        Object {
          "contributing": true,
          "statusSummary": "GOOD",
          "type": "LAG",
          "value": 5,
          "valuePresent": true,
        },
      ],
      "time": 2,
      "uuid": "2",
    },
    Object {
      "allStationAggregates": Array [],
      "channelSohs": Array [],
      "id": "3",
      "needsAcknowledgement": true,
      "needsAttention": true,
      "sohStatusSummary": "BAD",
      "stationGroups": Array [
        Object {
          "groupName": "Group 4",
          "sohStationCapability": "GOOD",
          "stationName": "H07N",
        },
      ],
      "stationName": "H07N",
      "statusContributors": Array [
        Object {
          "contributing": true,
          "statusSummary": "GOOD",
          "type": "LAG",
          "value": 5,
          "valuePresent": true,
        },
      ],
      "time": 2,
      "uuid": "3",
    },
    Object {
      "allStationAggregates": Array [],
      "channelSohs": Array [],
      "id": "4",
      "needsAcknowledgement": true,
      "needsAttention": true,
      "sohStatusSummary": "GOOD",
      "stationGroups": Array [
        Object {
          "groupName": "Group 4",
          "sohStationCapability": "MARGINAL",
          "stationName": "H08N",
        },
      ],
      "stationName": "H08N",
      "statusContributors": Array [
        Object {
          "contributing": true,
          "statusSummary": "GOOD",
          "type": "ENV_AUTHENTICATION_SEAL_BROKEN",
          "value": 5,
          "valuePresent": true,
        },
      ],
      "time": 2,
      "uuid": "4",
    },
  ],
}
`;

exports[`Soh station groups layout matches the snapshot when provided with a context 1`] = `
<div>
  <div
    class="resize-container 
        quadra-grid-container
        display__scroll-container--vertical
        drop-zone__wrapper"
  >
    <div
      class="resizer quadra-grid-cell soh-overview-table__wrapper"
      data-cy="soh-overview-group-Group 4"
      style="height: 360px; max-height: 100vh;"
    >
      <div
        class="divider-container"
      >
        <div
          class="top-component"
          style="height: 200px; min-height: 145px;"
        >
          <div
            class="soh-overview-table"
          >
            <div
              class="soh-overview-group-header__container"
              data-cy="soh-overview-header"
            >
              <div
                class="soh-overview-group-header__network"
                data-capability-status="bad"
                data-cy="soh-overview-group-header__network"
                title="Total station count for this group: 4"
              >
                Group 4
              </div>
              <div
                class="soh-overview-group-header__count--bad soh-overview-group-header__count"
                data-cy="soh-overview-group-header__count--bad"
              >
                0
              </div>
              <div
                class="soh-overview-group-header__count--marginal soh-overview-group-header__count"
                data-cy="soh-overview-group-header__count--marginal"
              >
                1
              </div>
              <div
                class="soh-overview-group-header__count--good soh-overview-group-header__count"
                data-cy="soh-overview-group-header__count--good"
              >
                3
              </div>
            </div>
            <div
              class="soh-overview-table__bin soh-overview-table__bin--top"
            >
              <div
                class="soh-table-label"
              >
                Needs Attention
              </div>
              <div
                class="soh-overview-cell__container"
                data-cy="soh-unacknowledged"
              >
                <div
                  draggable="true"
                >
                  <div
                    class="soh-overview-cell
          soh-overview-cell--draggable
          
        "
                    data-cy="soh-overview-cell"
                    data-cy-status="GOOD"
                    data-soh-status="good"
                    data-station-id="H05N"
                    tabindex="0"
                    title="Capability Rollup - GOOD"
                  >
                    H05N
                  </div>
                </div>
                <div
                  draggable="true"
                >
                  <div
                    class="soh-overview-cell
          soh-overview-cell--draggable
          
        "
                    data-cy="soh-overview-cell"
                    data-cy-status="GOOD"
                    data-soh-status="good"
                    data-station-id="H06N"
                    tabindex="0"
                    title="Capability Rollup - GOOD"
                  >
                    H06N
                    <div
                      class="soh-cell__right-container"
                      data-capability-status="marginal"
                      data-cy="soh-worst-of-badge"
                      style="width: 30px;"
                    >
                      <img
                        alt="MARGINAL"
                        class="badge"
                        height="16.88"
                        src="test-file-stub"
                        title="Worst of Rollup - MARGINAL"
                        width="18.996"
                      />
                    </div>
                  </div>
                </div>
                <div
                  draggable="true"
                >
                  <div
                    class="soh-overview-cell
          soh-overview-cell--draggable
          
        "
                    data-cy="soh-overview-cell"
                    data-cy-status="GOOD"
                    data-soh-status="good"
                    data-station-id="H07N"
                    tabindex="0"
                    title="Capability Rollup - GOOD"
                  >
                    H07N
                    <div
                      class="soh-cell__right-container"
                      data-capability-status="bad"
                      data-cy="soh-worst-of-badge"
                      style="width: 30px;"
                    >
                      <img
                        alt="BAD"
                        class="badge"
                        height="23"
                        src="test-file-stub"
                        title="Worst of Rollup - BAD"
                        width="17.583"
                      />
                    </div>
                  </div>
                </div>
                <div
                  draggable="true"
                >
                  <div
                    class="soh-overview-cell
          soh-overview-cell--draggable
          
        "
                    data-cy="soh-overview-cell"
                    data-cy-status="MARGINAL"
                    data-soh-status="marginal"
                    data-station-id="H08N"
                    tabindex="0"
                    title="Capability Rollup - MARGINAL"
                  >
                    H08N
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="horizontal-divider"
        >
          <div
            class="horizontal-divider__target"
          />
          <div
            class="horizontal-divider__spacer"
          />
          <svg
            class="resize-handle"
            height="7"
            viewBox="0 0 18 7"
            width="18"
          >
            <title>
              Resize
            </title>
            <g
              class="resize-handle-group"
              data-name="resize-handle-group"
            >
              <rect
                height="7"
                rx="1"
                width="2"
              />
              <rect
                height="7"
                rx="1"
                width="2"
                x="4"
              />
              <rect
                height="7"
                rx="1"
                width="2"
                x="8"
              />
              <rect
                height="7"
                rx="1"
                width="2"
                x="12"
              />
              <rect
                height="7"
                rx="1"
                width="2"
                x="16"
              />
            </g>
          </svg>
          <div
            class="horizontal-divider__spacer"
          />
        </div>
        <div
          class="bottom-component"
          style="height: 1px; min-height: 110px;"
        >
          <div
            class="soh-overview-table"
          >
            <div
              class="soh-overview-table__bin soh-overview-table__bin--bottom"
              data-cy="soh-acknowledged"
            >
              <div
                class="drop-zone
          
          soh-overview-table__drop-zone soh-overview-cell__container
                  soh-overview-table__drop-zone--highlighted"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="resizer__handle resizer__handle--height resizer__handle--bottom"
      />
      <div
        class="resizer__sensor resizer__sensor--bottom"
      />
    </div>
  </div>
</div>
`;

exports[`Soh station groups layout matches the snapshot when provided with no station groups to display 1`] = `
<div>
  <div
    class="resize-container 
        quadra-grid-container
        display__scroll-container--vertical
        drop-zone__wrapper"
  />
</div>
`;
